osu.Game/Online/Chat/MessageFormatter.cs

```diff
@@ -341,6 +341,8 @@ public enum LinkAction
         OpenWiki,
         Custom,
         OpenChangelog,
+        FilterBeatmapSetGenre,
+        FilterBeatmapSetLanguage,
```

Comment:
I do wonder a bit whether these should be permitted to be their own thing. You could change `SearchBeatmapSet` so that its argument is some sort of search query structure rather than just a query string, and then reuse that for this feature. It's probably a larger refactor though.

Not entirely sold on this though, so I may wait and see what others' opinions are.

Reply:
Also not completely sold, but adding here for now doesn't seem harmful. Refactoring can come in the future when required.